services/venice-controller/src/main/java/com/linkedin/venice/controller/ZkHelixAdminClient.java

```diff
@@ -98,6 +102,52 @@ public void createVeniceControllerCluster() {
         // Topology and fault zone type fields are used by CRUSH alg. Helix would apply the constrains on CRUSH alg to
         // choose proper instance to hold the replica.
         clusterConfig.setTopologyAwareEnabled(false);
+        clusterConfig.setPersistBestPossibleAssignment(true);
+
+        // We want to prioritize evenness over less movement when it comes to resource assignment, because the cost
+        // of rebalancing for the controller is cheap as it is stateless.
+        Map<ClusterConfig.GlobalRebalancePreferenceKey, Integer> globalRebalancePreference = new HashMap<>();
+
+        if (commonConfig.getHelixRebalancePreferenceEvenness() > -1
+            && commonConfig.getHelixRebalancePreferenceLessMovement() > -1) {
+          // EVENNESS and LESS_MOVEMENT need to be defined together
+          globalRebalancePreference.put(
+              ClusterConfig.GlobalRebalancePreferenceKey.EVENNESS,
+              commonConfig.getHelixRebalancePreferenceEvenness());
+          globalRebalancePreference.put(
+              ClusterConfig.GlobalRebalancePreferenceKey.LESS_MOVEMENT,
+              commonConfig.getHelixRebalancePreferenceLessMovement());
+        }
+
+        if (commonConfig.getHelixRebalancePreferenceForceBaselineConverge() > -1) {
+          globalRebalancePreference.put(
+              ClusterConfig.GlobalRebalancePreferenceKey.FORCE_BASELINE_CONVERGE,
+              commonConfig.getHelixRebalancePreferenceForceBaselineConverge());
+        }
+
+        if (!globalRebalancePreference.isEmpty()) {
+          clusterConfig.setGlobalRebalancePreference(globalRebalancePreference);
+        }
+
+        if (commonConfig.getHelixInstanceCapacity() > -1 && commonConfig.getHelixResourceCapacityWeight() > -1) {
+          List<String> instanceCapacityKeys = new ArrayList<>();
+          instanceCapacityKeys.add(CONTROLLER_DEFAULT_HELIX_RESOURCE_CAPACITY_KEY);
```

Comment:
nit: Can we change to `Collections.singletonList(CONTROLLER_DEFAULT_HELIX_RESOURCE_CAPACITY_KEY)` ?